tests/common/util.rs

```diff
@@ -1191,26 +1221,613 @@ impl UCommand {
+        expected_stderr_size: usize,
+    ) -> CmdResult {
+        self.with_output(AssertionMode::Exact(
+            expected_stdout_size,
+            expected_stderr_size,
+        ))
+    }
+
+    // Assert that the child process is alive
+    #[allow(clippy::wrong_self_convention)]
+    pub fn is_alive(&mut self) -> &mut Self {
+        match self
+            .uchild
+            .raw
+            .try_wait()
+        {
+            Ok(Some(status)) => panic!(
+                "Assertion failed. Expected '{}' to be running but exited with status={}.\nstdout: {}\nstderr: {}",
+                uucore::util_name(),
+                status,
+                self.uchild.stdout_all(),
+                self.uchild.stderr_all()
+            ),
+            Ok(None) => {}
+            Err(error) => panic!("Assertion failed with error '{:?}'", error),
+        }
+
+        self
+    }
+
+    // Assert that the child process has exited
+    #[allow(clippy::wrong_self_convention)]
+    pub fn is_not_alive(&mut self) -> &mut Self {
+        match self
+            .uchild
+            .raw
+            .try_wait()
+        {
+            Ok(None) => panic!(
+                "Assertion failed. Expected '{}' to be not running but was alive.\nstdout: {}\nstderr: {}",
+                uucore::util_name(),
+                self.uchild.stdout_all(),
+                self.uchild.stderr_all()),
+            Ok(_) =>  {},
+            Err(error) => panic!("Assertion failed with error '{:?}'", error),
+        }
+
+        self
+    }
+}
+
+/// Abstraction for a [`std::process::Child`] to handle the child process.
+pub struct UChild {
+    raw: Child,
+    bin_path: String,
+    util_name: Option<String>,
+    captured_stdout: Option<CapturedOutput>,
+    captured_stderr: Option<CapturedOutput>,
+    ignore_stdin_write_error: bool,
+    join_handle: Option<JoinHandle<io::Result<()>>>,
+    tmpd: Option<Rc<TempDir>>, // drop last
+}
+
+impl UChild {
+    fn new(
+        child: Child,
+        bin_path: String,
+        util_name: Option<String>,
+        tmpd: Option<Rc<TempDir>>,
+        captured_stdout: Option<CapturedOutput>,
+        captured_stderr: Option<CapturedOutput>,
+        ignore_stdin_write_error: bool,
+    ) -> Self {
+        Self {
+            raw: child,
+            bin_path,
+            util_name,
+            tmpd,
+            captured_stdout,
+            captured_stderr,
+            ignore_stdin_write_error,
+            join_handle: None,
+        }
+    }
+
+    /// Convenience method for `sleep(Duration::from_millis(millis))`
+    pub fn delay(&mut self, millis: u64) -> &mut Self {
+        sleep(Duration::from_millis(millis));
+        self
+    }
+
+    /// Return the pid of the child process, similar to [`Child::id`].
+    pub fn id(&self) -> u32 {
+        self.raw.id()
+    }
+
+    /// Returns true if the child process is still alive and false otherwise.
+    #[allow(clippy::wrong_self_convention)]
```

Comment:
I think these annotations can be removed, because the self convention is correct with the name change.

Reply:
I think you're mixing something up here. The wrong self convention comes with naming the method `is_...`.

Reply:
I can't reproduce it locally, so I'm confused. I'm running
```
cargo clippy --all-targets -- -D warnings
```

As far as I can tell from the source code of the clippy lint, it should allow `is_alive(&mut self)`, but only warns for `is_alive(self)`. So we should be good right? Or am I missing something?

Reply:
dunno. Here are the docs for this warning https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention. I'm getting this warning locally for `is_alive(&mut self)`

Reply:
I'm using `cargo clippy --features unix --test tests`

Reply:
The docs are why I'm confused. Here's a playground link that shows that mutable reference should be fine: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d0107deeee912829ef29bf6ef5385daa

In the last year, it appears this lint has been changed twice:
- Disallowing `&mut self`: https://github.com/rust-lang/rust-clippy/issues/8142, https://github.com/rust-lang/rust-clippy/pull/8208
- And then allowing it again: https://github.com/rust-lang/rust-clippy/issues/8513, https://github.com/rust-lang/rust-clippy/pull/8738

So maybe your clippy is out of date?

Reply:
I'm using the clippy version from the current min rust version 1.60.0 which is up-to-date  although it's not the latest version around

```
$ rustup show active-toolchain
1.60.0-x86_64-unknown-linux-gnu

$ cargo clippy --version
clippy 0.1.60 (7737e0b 2022-04-04)
```

Reply:
Oh that explains it I guess. The CI uses the latest stable version and we have a min rust version setting for it to ensure compatibility with msrv, so let's follow the CI.

Reply:
ok. The `#[allow...wrong_self_convention]` attributes are removed from the is_* methods.